pydrive2/auth.py

```diff
@@ -95,11 +95,8 @@ def _decorated(self, *args, **kwargs):
       dirty = True
     else:
       if self.access_token_expired:
-        if self.credentials.refresh_token is not None:
-          self.Refresh()
-        else:
-          decoratee(self, *args, **kwargs)
-          self.Authorize()
+        decoratee(self, *args, **kwargs)
```

Comment:
you probably don't need to call decoratee here at all, and then it means you don't to include that artificial if in the `ServiceAuth` code